c2corg_api/views/validation.py

```diff
@@ -624,18 +643,25 @@ def _add_associations(
             errors.add(
                 'body', 'associations.' + document_key,
                 'invalid association type')
-        else:
-            if document_key == 'waypoints' and main_document_type != BOOK_TYPE:
-                is_parent = True
-            elif document_key == 'waypoint_children':
-                is_parent = False
-
-            associations[document_key] = [
-                {
-                    'document_id': doc['document_id'],
-                    'is_parent': is_parent
-                } for doc in associations_in[document_key]
-                ]
+            return
+        
+        if not _are_valid_waypoints(associations_in, main_document_type, document_key):
+            errors.add(
+                'body', 
+                'associations.climbing_indoor_waypoint', 
+                'climbing_indoor waypoint cannot be linked to a route')
```

Comment:
this will require to add those translations for the frontend. I'll try not to forget it when merged.